### PR TITLE
Add warning about reference style links in shortcode inner text

### PR DESCRIPTION
**Pull request by jdbaldry**
Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

Reference style links such as `[link text][label]` or `[link text][]` do not work in the inner text of shortcodes.
For more information, refer to https://discourse.gohugo.io/t/markdown-reference-links-in-shortcodes/5770/3.

### CODE REVIEW

**Comment by clayton-cornell** [comment]
The workaround is to define the links as:

- A normal markdown formatted link
- A relref link with the relref definitions within the admonition
- Manually format the note/caution/warning using `>` block quotes

**Comment by jdbaldry** [comment]
> * A normal markdown formatted link
I think we should instead talk about inline vs. reference-style.
Neither is necessarily normal.

> * A relref link with the relref definitions within the admonition
Another point of technicality is that it's not so much of a relref link. It's an inline link using the `relref` shortcode to get the link destination.

You can for example also do:

```markdown
[link text][]

[link text]: {{< relref "<ARG>" >}}
```

> * Manually format the note/caution/warning using `>` block quotes
I'm not sure about recommending this as a workaround. I think consistent admonitions might be more important than supporting admonitions with links.

**Comment by clayton-cornell** [comment]
I've updated the bullet points...

File: docs/sources/write/shortcodes/index.md [hunk 1]
```diff
@@ -37,6 +37,11 @@ The content of the admonition must be within opening and closing tags.
 | --------- | --------------------------------------------------------------------- | -------- |
 | `type`    | The type of admonition. One of `"note"`, `"caution"`, or `"warning"`. | yes      |
 
+{{% admonition type="warning" %}}
+Reference style links such as `[link text][label]` or `[link text][]` do not work in the inner text of shortcodes.
+For more information, refer to https://discourse.gohugo.io/t/markdown-reference-links-in-shortcodes/5770/3.
```
Comment (jdbaldry) [inline docs/sources/write/shortcodes/index.md:42]:
```suggestion
For more information, refer to [Markdown Reference Links in Shortcodes](https://discourse.gohugo.io/t/markdown-reference-links-in-shortcodes/5770/3).
```

**Comment by clayton-cornell** [comment]
@jdbaldry I've got this PR lurking in my notifications. Is this one we can merge?

**Review by josmperez — APPROVED** [review]
LGTM.

**Pull request merged**